src/version.ts

```diff
@@ -1,6 +1,41 @@
 import { semver } from "./deps.ts";
 
-export class Version {
+export interface Version {
+  main: string;
+  app?: string;
+  prerelease: boolean;
+
+  compare(other: Version): number;
+  satisfies(spec: string): boolean;
+}
+
+const COMMIT_REGEX = /^[0-9a-f]{40}$/;
+
+export function makeVersion(main: string, app?: string, prerelease?: boolean) {
+  if (COMMIT_REGEX.test(main)) {
+    return new CommitVersion(main);
```

Comment:
Now that we support non-semantic versions, maybe it makes sense to not `semver.coerce` versions? Or at least treat versions that have been coerced differently. When I was initially working on this I just passed in the commit hash as the version and `semver` just coerced whatever number it could find in the version which is a bit iffy.